Tests/Factory/ZipkinTracerFactoryTest.php

```diff
@@ -105,4 +126,20 @@ public function testCreateException(): void
             )
         );
     }
+
+    /**
+     * @return array{endpoint_url: string}
+     * @throws \ReflectionException
+     */
+    private function extractReporterOptionsByReflection(\OpenTracing\Tracer $tracer)
+    {
+        $tracerRef = new \ReflectionClass($tracer);
+        $ottTracer = $tracerRef->getProperty('tracer')->getValue($tracer);
+        $ottTracerRef = new \ReflectionClass($ottTracer);
+        $recorder = $ottTracerRef->getProperty('recorder')->getValue($ottTracer);
+        $recorderRef = new \ReflectionClass($recorder);
+        $reporter = $recorderRef->getProperty('reporter')->getValue($recorder);
+        $reporterRef = new \ReflectionClass($reporter);
+        return $reporterRef->getProperty('options')->getValue($reporter);
```

Comment:
```suggestion
        $tracerRef = new \ReflectionClass($tracer);
        $tracerProperty = $tracerRef->getProperty('tracer');
        $tracerProperty->setAccessible(true);
        $ottTracer = $tracerProperty->getValue($tracer);
        $ottTracerRef = new \ReflectionClass($ottTracer);
        $recorderProperty = $ottTracerRef->getProperty('recorder');
        $recorderProperty->setAccessible(true);
        $recorder = $recorderProperty->getValue($ottTracer);
        $recorderRef = new \ReflectionClass($recorder);
        $reporterProperty = $recorderRef->getProperty('reporter');
        $reporterProperty->setAccessible(true);
        $reporter = $reporterProperty->getValue($recorder);
        $reporterRef = new \ReflectionClass($reporter);
        $optionsProperty = $reporterRef->getProperty('options');
        $optionsProperty->setAccessible(true);
        return $optionsProperty->getValue($reporter);
```

This test is failing in some configurations, although I don't understand the reason completely. However, explicitly allowing access to the private properties solves this issue.

If you have a cleaner solution to this, you are welcome to implement it instead.